src/resources/js/customizer-views-v2-live-preview.js

```diff
@@ -0,0 +1,832 @@
+/* global tribe, tribe_events_customizer_live_preview_js_config */
+
+/**
+ * File customizer-views-v2-live-preview.js.
+ *
+ * Plugin Customizer enhancements for a better user experience.
+ *
+ * Contains handlers to make TEC Customizer preview reload changes asynchronously.
+ *
+ * Please, for sanity's sake - try to keep controls organized by how they appear in the customizer!
+ */
+
+var tribe_events_customizer_live_preview_js_config = tribe_events_customizer_live_preview_js_config || {};
+
+( function( $, api, obj ) {
```

Comment:
This whole JS file does a lot. A couple thoughts:

1. I worry that we're duplicating a **lot** of code here. This is the same logic we do in the BE, and when we add new customizer functionality in the PHP, we also need to update the JS here, which ends up being fairly brittle. Maybe we should think about how to do this without duplicating here?
2. If we keep this, I wonder if there's a way to simplify some of this logic. I see that there is some that require a bit more logic, but there's a lot that has similar logic.